internal/daemon/daemon.go

```diff
@@ -0,0 +1,136 @@
+package daemon
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/aikido/sc-agent/internal/proxy"
+	"github.com/aikido/sc-agent/internal/scannermanager"
+	"github.com/aikido/sc-agent/internal/version"
+)
+
+type Config struct {
+	ConfigPath string
+	LogLevel   string
+}
+
+type Daemon struct {
+	config   *Config
+	ctx      context.Context
+	cancel   context.CancelFunc
+	wg       sync.WaitGroup
+	stopOnce sync.Once
+	proxy    *proxy.Proxy
+	registry *scannermanager.Registry
+}
+
+func New(config *Config) (*Daemon, error) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
You don't need this context, everywhere else you're using the one that's passed in with Start/Stop.